src/structure/spatial.jl

```diff
@@ -186,10 +189,25 @@ function omegaangle(res::AbstractResidue, res_prev::AbstractResidue)
     return dihedralangle(res_prev["CA"], res_prev["C"], res["N"], res["CA"])
 end
 
+function omegaangle(chain::Chain, res_id::Union{Integer, AbstractString})
+    inds = find(r -> r == string(res_id), resids(chain))
+    if length(inds) != 1
+        throw(ArgumentError("$res_id is an invalid residue ID"))
+    end
+    i = inds[1]
+    res = chain[resids(chain)[i]]
```

Comment:
I just realised that this needs re-ordering to catch the error properly, at the minute if i is 1 it fails on line 199. Probably easiest to remove lines 198 and 199 and switch lines 200-203 to:
```julia
    if i == 1 || !sequentialresidues(chain[resids(chain)[i-1]], chain[resids(chain)[i]])
        throw(ArgumentError("$res_id is an invalid residue ID"))
    end
    return omegaangle(chain[resids(chain)[i]], chain[resids(chain)[i-1]])
```
and similar for the other 2 functions.